File: app/src/main/java/workshop/akbolatss/tagsnews/screen/sources/SourcesView.java
package workshop.akbolatss.tagsnews.screen.sources;

import java.util.List;

import workshop.akbolatss.tagsnews.base.BaseView;
import workshop.akbolatss.tagsnews.base.ErrorView;
import workshop.akbolatss.tagsnews.base.LoadingView;
import workshop.akbolatss.tagsnews.repositories.source.RssSource;

public interface SourcesView extends BaseView, LoadingView, ErrorView {

    void onLoadSources(List<RssSource> rssSourceList);

    void onAddNewSource();
}

File: app/src/main/java/workshop/akbolatss/tagsnews/screen/sources/SourcesView.kt
package workshop.akbolatss.tagsnews.screen.sources

import workshop.akbolatss.tagsnews.base.BaseView
import workshop.akbolatss.tagsnews.base.ErrorView
import workshop.akbolatss.tagsnews.base.LoadingView
import workshop.akbolatss.tagsnews.model.dao.RssSource

/**
 * MVP View for #SourcesView
 * @see SourcesActivity
 */
interface SourcesView : BaseView, LoadingView, ErrorView {

    fun onLoadSources(rssSourceList: List<RssSource>)

    fun onAddNewSource()
}
